cognite/client/data_classes/contextualization.py

```diff
@@ -539,6 +539,105 @@ class PersonalProtectiveEquipmentDetectionParameters(VisionResource, ThresholdPa
+        if not job_ids:
+            raise ValueError("You need to specify job_ids")
+        # TODO: Can maybe improve this and/or generalize
+        self.job_ids = job_ids
+        self.jobs: List[Dict[str, Any]] = []
+        self.succeeded: List[Dict[str, Any]] = []
+        self.failed: List[Dict[str, Any]] = []
+        self._result: List[Dict[str, Any]] = []
+        self._remaining_job_ids: List[Dict[str, Any]] = []
+
+    def __repr__(self) -> str:
+        return (
+            "DetectJobBundle( job_ids="
+            + str(self.job_ids)
+            + ", self.jobs="
+            + str(self.jobs)
+            + ", self._result="
+            + str(self._result)
+            + ", self.failed="
+            + str(self.failed)
+            + ", self.succeeded="
+            + str(self.succeeded)
+            + ", self._remaining_job_ids="
+            + str(self._remaining_job_ids)
+            + ")"
+        )
+
+    def back_off(self) -> None:
+        """
+        Linear back off, in order to limit load on our API.
+        Starts at _WAIT_TIME and goes to 10 seconds.
+        """
+        time.sleep(self._WAIT_TIME)
+        if self._WAIT_TIME < 10:
+            self._WAIT_TIME += 2
+        return
+
+    def wait_for_completion(self, timeout: int = None) -> None:
+        """Waits for all jobs to complete, generally not needed to call as it is called by result.
+        Args:
+            timeout (int): Time out after this many seconds. (None means wait indefinitely)
+            interval (int): Poll status every this many seconds.
+        """
+        start = time.time()
+        # Create smart procedure for pinging less and less. One DB call for all.
+        c = 0
+        while timeout is None or time.time() < start + timeout:
+            if c == 0:
+                self.jobs = self._cognite_client.diagrams._post(self._STATUS_PATH, json={"items": self.job_ids}).json()[
+                    "items"
+                ]
+            elif len(self._remaining_job_ids) >= 1:
+                self.back_off()
+                self.jobs = self._cognite_client.diagrams._post(
+                    self._STATUS_PATH, json={"items": self._remaining_job_ids}
+                ).json()["items"]
+            else:
+                return
+            # Assign the jobs that aren't finished
+            self._remaining_job_ids = [j["jobId"] for j in self.jobs if JobStatus(j["status"]).is_not_finished()]
+            c += 1
+
+    def fetch_results(self) -> List[Dict[str, Any]]:
+        # Reset
+        self.failed = []
+        self.succeeded = []
+
+        # Check status
+        list_of_job_results = [
+            self._cognite_client.diagrams._get(f"{self._RESOURCE_PATH}{j}").json() for j in self.job_ids
+        ]
+
+        # Break out
+        for job_result in list_of_job_results:
+            for item in job_result["items"]:
+                if "errorMessage" in item:
```

Comment:
One ugly fact is that we dont return anything at all on jobs that time out in the worker, the number of items will be lower than what was sent in... This should be fixed on the API side though.

Reply:
Will look into a way to check the number of returned vs. posted

Reply:
Even timed out jobs will return as failed, so there shouldn't be a problem here.